src/main/java/com/hierynomus/spnego/NegTokenInit.java

```diff
@@ -102,11 +103,18 @@ public NegTokenInit read(byte[] bytes) throws IOException {
     public NegTokenInit read(Buffer<?> buffer) throws IOException {
         try {
             ASN1Primitive applicationSpecific = new ASN1InputStream(buffer.asInputStream()).readObject();
-            if (!(applicationSpecific instanceof ASN1ApplicationSpecific)) {
+            if (!(applicationSpecific instanceof BERApplicationSpecific || applicationSpecific instanceof DERApplicationSpecific)) {
```

Comment:
Why do we need compat with earlier BC?

Reply:
Is there any chance of replacing ASN.1 from BouncyCastle with something else? BouncyCastle dependency is a bit problematic because of US strong cryptography export restrictions: https://en.wikipedia.org/wiki/Strong_cryptography#Legal_issues

We have solved this by making our users download BC separately and add it to our application classpath, but it complicates the installation a bit.

What's the current status of https://github.com/hierynomus/asn-one?

Reply:
Indeed, asn-one is underway. Need to put some focus time on it to finish it. Can't even remember what was holding it back 😉 ...
Feel free to submit a PR to migrate to asn-one based on the current master build. Then we can see quickly enough what's still missing!

Reply:
@krivanekm-javlin Just created PR #394. It is working for the integration tests. Just need to release a final version of asn-one now.

Reply:
Wow, that was fast! I started working on it and I think I have the changes ready, but you were faster. :)

What does this mean? Will BC still be a required dependency? There are references in com.hierynomus.security.bc package. Are those required or optional at runtime?

In the case it's still required:
Since Java 8u161, unlimited cryptography is enabled by default:
https://www.oracle.com/technetwork/java/javase/8u161-relnotes-4021379.html#JDK-8170157 Would this allow making BouncyCastle optional when running with unlimited cryptography?

Thanks for your reply.

Reply:
@krivanekm-javlin Let's continue the discussion on PR #394.